### ISSUE TITLE
Add assert harness around unsafe direct memory access

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Map#335]

**Comment by minborg** [comment]
This can be done using the zero-cost assertion mechanism in chronicle-core, e.g.:
```
    @Override
    public long readEntry(long addr, long pos) {
        assert SKIP_ASSERTIONS || addr != 0 && addr != -1 : "addr: " + addr;
        assert SKIP_ASSERTIONS || assertIfEnabled(nonNegative(), pos) : "pos is negative";
        return OS.memory().readInt(addr + pos);
    }
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.22ea3](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.22ea3), [BOM-2.22ea23](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea23)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.22ea3](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.22ea3), [BOM-2.22ea29](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea29)
